source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+nec tortor nec arcu molestie dapibus. Phasellus at nunc ut quam molestie
+finibus. Suspendisse finibus lectus eget nulla eleifend, vel consequat
+leo rhoncus. Phasellus lacinia tortor id aliquet condimentum. Morbi
+sodales est a leo rhoncus, non pretium neque dignissim.
+
+Solution
+~~~~~~~~
+
+Praesent sit amet elit pretium mi molestie porta a sit amet ipsum. Sed
+sollicitudin purus et mi blandit sodales.
+
+- Deterministic vs Non-Deterministic
+
+Procedure
+---------
+
+Requirements
+~~~~~~~~~~~~
+
+- Point to setup and install directions
+- Nullam imperdiet lorem vitae vulputate lacinia.
+- Donec eget velit tincidunt, gravida diam ac, efficitur lacus.
+
+A. Create a Local Master Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Aenean eu consequat lorem. Ut posuere est sed sodales pharetra. Cras
+volutpat, massa laoreet varius dictum, leo odio porttitor ante, nec
+auctor tortor orci et mi. Maecenas tempor, lacus vehicula molestie
+pulvinar, ante eros faucibus odio, sed consequat quam tellus vel arcu.
+Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere
+cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+
+B. Define a JSON Schema
+~~~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB uses JSON Schema to to validate data which is written to a collection.
+Within the context of Field Level Encryption, we use JSON Schema to specify several
+configuration details. For instance, we can specify the form of encryption we want to use 
+- deterministic or random. We can also specify the BSON Type of the field we are encrypting. 
+In addition, we can specify the key we want to use. This is particularly important, because
+it means that we can use different keys for different fields.
+
+In the following example, we will go over creation of a schema for a collection which represents patient data
+in a medical office. Some fields use deterministic encryption, and others use random (probabilistic encryption). 
+
+The structure for our schema will be as follows:
+
+.. code::
+
+    fullName: String
+    address: Encrypted String using random encryption
+    ssn: Encrypted Int using deterministic encryption
+    last4SSN: Int
+    patientInfo: Object
+    patientInfo.phone Encrypted Int using deterministic encryption
+    patientInfo.provider String
+    medicalRecords: Encrypted array of documents using random encryption
+
+Explanation of encryption choices:
+``````````````````````````````````
+* address is randomly encrypted because it is more secure for strings to be randomly encrypted to prevent hackers from decrypting by building a cipher-dictionary
+* ssn is deterministically encrypted because deterministic encryption provides filtering ability 
+* patientInfo.phone is deterministically encrypted for the same reason as ssn, and also it provides an example of encryption with a nested document
+* medicalRecords are randomly encrypted because entire array's are only encryptable with random encryption
```

Comment:
medicalRecords ~are~ is (referring to the field, singular)

array's -> arrays

I'm not sure of the use of the word "entire" here.

Reply:
I switched "array's" to "arrays". As for the second suggestion, "entire" makes sense technically. This is because, technically, array items are encryptable just not via automatic encryption with the driver. For instance, if I wanted to add the item "hello world" and encrypt it manually myself and push it into an unencrypted array, that is completely possible. But this example is going over auto encrypting the entire array.

Reply:
Might be a little more direct/clear to say "array fields cannot be encrypted deterministically"

i.e. array field vs array item/element

Reply:
Good choice @nlarew, fixed in the latest commit